File: python/p012.py
from math import sqrt

def triangles():
    i = 1
    sum = 0
    while True:
        sum += i
        i += 1
        yield sum

def divisors(n):
    count = 1
    for i in xrange(1, int(sqrt(n))):
        if n % i == 0:
            count += 2

    return count

for i in triangles():
    if i % 2 == 1:
        continue

    num_divisors = divisors(i)
    if num_divisors > 500:
        print i
        break

File: go/p012.go
package main

import "fmt"

func main() {

    for i, n := 1, 1; ; i, n = i + 1, n + i + 1 {
        if n % 2 == 1 {
            continue
        }

        if divisors(n) > 500 {
            fmt.Println(n)
            break
        }
    }
}

func divisors(n int) (int) {
    count := 0
    for i := 1; i * i < n; i++ {
        if n % i == 0 {
            count += 2
        }
    }
    return count
}
